appendix/single-sign-on.rst

```diff
@@ -0,0 +1,521 @@
+
+2f. Configure Apache
+^^^^^^^^^^^^^^^^^^^^
+
+Zammad provides a sample virtual host configuration file for Apache.
+Add it to your ``sites-available`` directory and enable it:
+
+.. code-block:: sh
+
+   # Ubuntu, Debian, openSUSE
+   $ cp /opt/zammad/contrib/apache2/zammad_ssl.conf /etc/apache2/sites-available/
+   $ chown www-data:www-data /etc/apache2/sites-available/zammad_ssl.conf
+   $ a2ensite zammad_ssl
+
+   # CentOS
+   $ cp /opt/zammad/contrib/apache2/zammad_ssl.conf /etc/httpd/sites-available/
+   $ chown apache:apache /etc/httpd/sites-available/zammad_ssl.conf
+   $ ln -s /etc/httpd/sites-available/zammad_ssl.conf /etc/httpd/sites-enabled/
+
+Also, make sure the following line is present in your Apache configuration:
+
+.. code-block::
+
+   # /etc/apache2/apache2.conf (Ubuntu, Debian, & openSUSE)
+   # /etc/httpd/conf/httpd.conf (CentOS)
+
+   IncludeOptional sites-enabled/*.conf
+
+Now that ``zammad_ssl.conf`` is in place,
+it must be modified for your server.
+Replace all instances of ``example.com`` with your Zammad FQDN,
+and add the following directive to the end of the file
+to create your Kerberos SSO endpoint at ``/auth/sso``:
+
+.. note:: Replace the following placeholders in the command below:
+
+   :``<zammad-host>``: Zammad FQDN
+   :``<domain>``:      Windows domain
+
+   The configuration below contains two ``Krb5KeyTab`` lines!
+   Keep only the one you need.
+
+.. code-block:: apache
+
+   <LocationMatch "/auth/sso">
+      SSLRequireSSL
+      AuthType Kerberos
+      AuthName "Your Zammad"
+      KrbMethodNegotiate On
+      KrbMethodK5Passwd On
+      KrbAuthRealms <domain>
+      KrbLocalUserMapping on                 # strips @REALM suffix from REMOTE_USER variable
+      KrbServiceName HTTP/<zammad-host>@<domain>
+      Krb5KeyTab /etc/apache2/zammad.keytab  # Ubuntu, Debian, & openSUSE
+      Krb5KeyTab /etc/httpd/zammad.keytab    # CentOS
+      require valid-user
+
+      RewriteEngine On
+      RewriteCond %{LA-U:REMOTE_USER} (.+)
+      RewriteRule . - [E=RU:%1,NS]
+      RequestHeader set X-Forwarded-User "%{RU}e" env=RU
+   </LocationMatch>
+
+2g. Restart Apache to apply changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: sh
+
+   $ systemctl restart apache2
+
+Step 3: Enable SSO in Zammad
+----------------------------
+
+Next, enable “Authencation via SSO” in Zammad’s Admin Panel under **Settings > Security > Third-Party Applications**:
+
+.. figure:: /images/appendix/single-sign-on/authentication-via-sso.png
+   :align: center
+   :alt: “Authentication via SSO” toggle button in the Admin Panel
+
+   In Zammad 3.5, this option
+   adds a **Sign in using SSO** button to the sign-in page.
+
+.. note:: 
+   On older versions of Zammad,
+   visit ``https://your.zammad.host/auth/sso`` to sign in.
+
+Step 4: Configure Client System (Windows Only)
+----------------------------------------------
+
+Finally, Zammad users on the Active Directory server’s local intranet
```

Comment:
Is "local intranet" the right word here?

Reply:
Yes!
It's he actual windows wording for intranet sites.

![image](https://user-images.githubusercontent.com/6549061/94598735-ad89cc00-028f-11eb-9780-a506a08a8302.png)

Adding the fqdn of the zammad server (or whatever you use to call Zammads webpage) has to be configured as a local intranet site. This tells the users browser that it can trust the page in question and run the kerberos authentication against that page.

Other wise the browsers won't

Reply:
Got it. Just to make sure I understand you:

1. If you don't take this configuration step, you can still use SSO to sign in, but you have to manually enter your Active Directory username and password.
2. If the Zammad server is actually on the local intranet, you still have to do this.

Are the above points correct?

Reply:
Yes, both points are correct.
Sometimes the environments are configured to "detect" the intranet pages automatically, but I wouldn't cover that or rather ignore it..

Reply:
I've just pushed the following revision:

![Screenshot_20201002_222145](https://user-images.githubusercontent.com/12194123/94934020-c3042d80-04fd-11eb-9e9d-7472d4799dc1.png)

This wording okay with you?

Reply:
Yep, that's fine to me! Thanks!